libs/agno/agno/os/auth.py

```diff
@@ -55,3 +58,175 @@ def validate_websocket_token(token: str, settings: AgnoAPISettings) -> bool:
 
     # Verify the token matches the configured security key
     return token == settings.os_security_key
+
+
+def require_scopes(required_scopes: List[str]):
+    """
+    Create a FastAPI dependency that checks for required scopes.
+
+    This is a helper function for future use when you want to add
+    scope requirements directly to specific endpoints using FastAPI's
+    Depends() mechanism.
+
+    Usage:
+        @router.get("/endpoint", dependencies=[Depends(require_scopes(["scope:read"]))])
+        async def my_endpoint():
+            ...
+
+    Args:
+        required_scopes: List of scopes required to access the endpoint
+
+    Returns:
+        A dependency function that validates scopes
+    """
+
+    def scope_checker(request: Request) -> bool:
+        # Check if user is authenticated
+        if not getattr(request.state, "authenticated", False):
+            raise HTTPException(
+                status_code=401,
+                detail="Authentication required",
+            )
+
+        # Get user's scopes from request state (set by JWT middleware)
+        user_scopes = getattr(request.state, "scopes", [])
+
+        # Check for admin scope (grants all permissions)
+        if "admin" in user_scopes:
```

Comment:
should this rather be `"agent_os:admin"` ? Maybe we should make enums for these like `AgentOSScope.ADMIN.value`?